lib/recurly/base.php

```diff
@@ -112,6 +112,27 @@ protected static function _delete($uri, $client = null)
     return null;
   }
 
+  /**
+   * Returns URI for resource, throwing error if resource code is an empty string
+   * @param string First Recurly Client path
+   * @param string Resource codes and paths
+   * @return string URI
+   * @throws Recurly_Error
+   */
+  public static function _safeUri(...$params) {
+    $path = "";
+    foreach($params as $string) {
+      if (empty(trim($string))) {
+        throw new Recurly_Error("Resource code cannot be an empty value");
+      }
+      // removes extra forward slash that is encoded when a second path constant is passed
+      // example: const PATH_ACCOUNTS = '/accounts';
+      $path .= '/' . preg_replace("(%2F)", "", rawurlencode($string));
```

Comment:
So this means that no path components can contain a slash correct? and if they do, they will be silently stripped?

Reply:
maybe change it so it only targets the first character. also is there a reason you're replacing after encoding rather than before?

Reply:
As a reader of the code, it's not immediately clear that you're replacing the slash other than the comment.